src/pages/MangaInfo.js

```diff
@@ -1,48 +1,44 @@
 // @flow
-import * as React from 'react';
-import type { MangaInfoContainerProps } from 'containers/MangaInfoContainer';
-import type { MangaType } from 'types';
-import { Helmet } from 'react-helmet';
-import MangaInfoHeader from 'components/mangaInfo/MangaInfoHeader';
-import MangaInfoDetails from 'components/mangaInfo/MangaInfoDetails';
-import FullScreenLoading from 'components/loading/FullScreenLoading';
-import MangaInfoChapters from 'components/mangaInfo/MangaInfoChapters';
+import * as React from "react";
+import type { MangaInfoContainerProps } from "containers/MangaInfoContainer";
+import { Helmet } from "react-helmet";
+import MangaInfoHeader from "components/mangaInfo/MangaInfoHeader";
+import MangaInfoDetails from "components/mangaInfo/MangaInfoDetails";
+import FullScreenLoading from "components/loading/FullScreenLoading";
+import MangaInfoChapters from "components/mangaInfo/MangaInfoChapters";
 
 type State = { tabValue: number };
 
 class MangaInfo extends React.Component<MangaInfoContainerProps, State> {
   state = {
-    tabValue: this.props.defaultTab,
+    tabValue: this.props.defaultTab
   };
 
   componentDidMount() {
     const {
-      fetchMangaInfo, fetchChapters, updateMangaInfo, updateChapters,
+      fetchMangaInfo,
+      fetchChapters,
+      updateMangaInfo,
+      updateChapters
     } = this.props;
 
-    fetchChapters()
-      .then(() => {
-        // Fetch chapters cached on the server
-        // If there are none, tell the server to scrape chapters from the site
-        if (!this.props.chapters.length) {
-          return updateChapters(); // return promise so next .then()'s wait
-        }
-        return null;
-      })
-      .then(() => fetchMangaInfo())
-      .then(() => {
-        // If we think the server hasn't had enough time to scrape the source website
-        // for this mangaInfo, wait a little while and try fetching again.
-        //
-        // NOTE: This only updates the manga being viewed. Many of your other search results are
-        //       likely missing information as well. Viewing them will then fetch the data.
-        //
-        // TODO: might try to do one additional fetch at a slightly later time. e.g. 1000 ms
-        const { mangaInfo } = this.props;
-        if (mangaInfo && possiblyMissingInfo(mangaInfo)) {
-          setTimeout(() => updateMangaInfo(), 300);
-        }
-      });
+    fetchChapters().then(() => {
+      // Fetch chapters cached on the server
+      // If there are none, tell the server to scrape chapters from the site
+      if (!this.props.chapters.length) {
+        return updateChapters(); // return promise so next .then()'s wait
+      }
+      return null;
+    });
+
+    fetchMangaInfo().then(() => {
+      const { mangaInfo } = this.props;
+      // Update manga manually if the server has not fetched the data for
+      //   this manga yet
+      if (mangaInfo && !mangaInfo.initialized) {
+        updateMangaInfo();
```

Comment:
Are you sure this will work without the setTimeout?
I remember when I was developing this part that the delay was necessary for the info to fetch.

Reply:
I'm not sure why the extra time was needed in the first place. The endpoint being called by `fetchMangaInfo` is only a couple lines of code: https://github.com/TachiWeb/TachiWeb-Server/blob/4a6b3335191b08b6ad8d2db4ce6de88ce9c3179c/TachiServer/src/main/java/xyz/nulldev/ts/api/http/manga/MangaRoute.kt#L36-L41 . All it does is get the manga from the database and serialize it into a JSON. It doesn't do any kind of scraping...

The v3 version of this endpoint that I've moved to does the exact same thing.

Either way, I didn't notice any problems after I made this change so it could have been a bug that was specific to the v1/v2 endpoints...

Reply:
Not sure if this helps, I believe I faced this problem when I started searching something new in the catalogue and tried to view a manga really fast. My guess is that somewhere in that chain of events the server is spending time scraping the source website, and because that takes time, it's possible for `fetchMangaInfo` to return almost no information

Reply:
Hmm, I've tried to replicate this but I still can't. I actually did rework most of the async/threading portions of the code from API v1/v2 to v3 so that's why I said maybe it was a bug specific to the v1/v2 endpoints.

I think we should leave the delay out for now and if it comes up again, we'll throw it back in.